### PR TITLE
feat(iot-service): Add proxy support for registry manager

### PR DESCRIPTION
**Pull request by timtay-microsoft**
Not for service client, just for registry manager

see #766

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java [hunk 1]
```diff
@@ -41,6 +44,10 @@
     private ExecutorService executor;
     private IotHubConnectionString iotHubConnectionString;
 
+    @Setter
```
Comment (drwill-ms) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java:47]: Is there a standard order in which to list these? I'd have assumed get would be first, if there was.

Comment (timtay-microsoft) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/RegistryManager.java:47]: There isn't a standard that I know of, unlike in C#. I can switch them if it's bugging you though

**Pull request merged**